app/models/authentication.rb

```diff
@@ -0,0 +1,11 @@
+class Authentication
+
+	def self.visit(code)
```

Comment:
also - i don't know if i love the name "visit" - can we think of anything else that describes what this is doing better?